rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java

```diff
@@ -931,12 +944,14 @@ public PickResult pickSubchannel(PickSubchannelArgs args) {
         if (picker == null) {
           return PickResult.withNoResult();
         }
+        // Happy path
         return picker.pickSubchannel(args);
       } else if (response.hasError()) {
         if (hasFallback) {
           return useFallback(args);
         }
-        return PickResult.withError(response.getStatus());
+        return PickResult.withError(
+            convertRlsServerStatus(response.getStatus(), request.keyMap().get("server")));
```

Comment:
Where did this "server" key come from? I think that's wrong in two ways:
1. That is defined by the RLS config, and is just a convention used in our tests from what I can see
2. The RLS server itself would know who it is. This server value here is the name for the backends, not the RLS server

We want `RouteLookupConfig.lookupService()`

Reply:
Thanks for the clarification.  You mentioned using lookupService and I misinterpreted it since it was coming from the lbPolicyConfig and I was expecting something on the request or response.